mise.toml

```diff
@@ -0,0 +1,62 @@
+#:schema https://mise.jdx.dev/schema/mise.json
+
+[tools]
+uv = "0.5.1"
+
+[tasks."setup"]
+description = "Set up the dev environment (install Rust, etc)"
+run = [
+  "rustup toolchain install 1.84",
+  # We want rustfmt from nightly
+  "rustup toolchain install nightly --profile minimal --component rustfmt",
+  # Tools installed via Cargo
+  "cargo +1.84 install --locked bacon",
+  "cargo +1.84 install --locked cargo-insta",
+  "cargo +1.84 install --locked cargo-nextest",
+]
+
+[tasks."format"]
+description = "Format the code"
+run = "cargo +nightly fmt"
+
+[tasks."lint"]
+description = "Lint the code"
+run = "cargo +1.84 clippy --workspace --all-targets"
+
+[tasks."lint:ci"]
+description = "Lint the code using the same clippy version as CI would"
```

Comment:
comment (non-blocking): If we wanted, we could also literally use `mise` in CI 👀. You could use https://github.com/jdx/mise-action to cache all of the tool versions as well.

Reply:
As per https://github.com/jj-vcs/jj/pull/6081/files#r2020168644, we don't want/need to use `mise` in CI directly.